crates/criticalup-cli/src/commands/clean.rs

```diff
@@ -16,6 +17,8 @@ pub(crate) async fn run(ctx: &Context) -> Result<(), Error> {
 
     delete_cache_directory(&ctx.config.paths.cache_dir).await?;
     delete_unused_installations(installations_dir, &state).await?;
+    // Deletes unused binary proxies after state cleanup.
```

Comment:
I updated this in my change, could you validate it is correct and working on Linux/Mac? (Also, maybe we should check that the `proxy_binary` arg passed to `binary_proxies::update` is actually a file not a directory...)